core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -209,6 +210,42 @@ class ZkUtils(val zkClient: ZkClient,
     }
   }
 
+
+  def getClusterId(): Option[String] = {
+    def parseClusterId(clusterIdString: String): String = {
+
+      Json.parseFull(clusterIdString) match {
+        case Some(m) =>
+          val clusterIdMap = m.asInstanceOf[Map[String, Any]]
+          clusterIdMap.get("id").get.asInstanceOf[String]
+        case None => throw new KafkaException("Failed to parse the cluster id json [%s].".format(clusterIdString))
+      }
+    }
+
+    readDataMaybeNull(ClusterIdPath)._1 match {
+      case Some(clusterId) => Some(parseClusterId(clusterId))
+      case None => None
+    }
+  }
+
+  def createOrGetClusterId(proposedClusterId: String): String = {
+    val jsonMap = Map("version" -> "1",
+      "id" -> proposedClusterId
+    )
```

Comment:
Maybe we can have an inner class that has both `toJson` and `fromJson` methods so that the logic for this is together. Also, it's worth adding a comment for the JSON format and do a simple roundtrip test.